cmd/tink-worker/internal/registry.go

```diff
@@ -64,8 +74,18 @@ func (r *RegistryConnDetails) pullImage(ctx context.Context, cli *client.Client,
 		return errors.Wrap(err, "DOCKER PULL")
 	}
 	defer out.Close()
-	if _, err := io.Copy(os.Stdout, out); err != nil {
-		return err
+	fd := json.NewDecoder(out)
+	var status *ImagePullStatus
+	for {
+		if err := fd.Decode(&status); err != nil {
+			if err == io.EOF {
+				break
+			}
+			return errors.Wrap(err, "DOCKER PULL")
+		}
+		if status.Error != "" {
+			return errors.Wrap(errors.New(status.Error), "DOCKER PULL")
+		}
```

Comment:
As we discussed on Slack, if we add `fmt.Printf("%+v\n", status)` below this line, we retain logging functionality and everything should work nicely then.

Reply:
Not relevant anymore. See #451.